cpu/kinetis_common/uart.c

```diff
@@ -0,0 +1,348 @@
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            UART_1_DEV->C2 &= ~(1 << UART_C2_TIE_SHIFT);
+            break;
+#endif
+
+        default:
+            break;
+    }
+}
+
+int uart_write(uart_t uart, char data)
+{
+    switch (uart) {
+#if UART_0_EN
+
+        case UART_0:
+            if (UART_0_DEV->S1 & UART_S1_TDRE_MASK) {
+                UART_0_DEV->D = (uint8_t)data;
+            }
+
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            if (UART_1_DEV->S1 & UART_S1_TDRE_MASK) {
+                UART_1_DEV->D = (uint8_t)data;
+            }
+
+            break;
+#endif
+
+        default:
+            return -2;
+            break;
+    }
+
+    return 0;
+}
+
+int uart_read_blocking(uart_t uart, char *data)
+{
+    switch (uart) {
+#if UART_0_EN
+
+        case UART_0:
+            while (!(UART_0_DEV->S1 & UART_S1_RDRF_MASK));
+
+            *data = (char)UART_0_DEV->D;
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            while (!(UART_1_DEV->S1 & UART_S1_RDRF_MASK));
+
+            *data = (char)UART_1_DEV->D;
+            break;
+#endif
+
+        default:
+            return -2;
+            break;
+    }
+
+    return 1;
+}
+
+int uart_write_blocking(uart_t uart, char data)
+{
+    switch (uart) {
+#if UART_0_EN
+
+        case UART_0:
+            while (!(UART_0_DEV->S1 & UART_S1_TDRE_MASK));
+
+            UART_0_DEV->D = (uint8_t)data;
+            break;
+#endif
+#if UART_1_EN
+
+        case UART_1:
+            while (!(UART_1_DEV->S1 & UART_S1_TDRE_MASK));
+
+            UART_1_DEV->D = (uint8_t)data;
+            break;
+#endif
+
+        default:
+            return -2;
+            break;
+    }
+
+    return 1;
+}
+
+#if UART_0_EN
+void UART_0_ISR(void)
+{
+    irq_handler(UART_0, UART_0_DEV);
+}
+#endif
+
+#if UART_1_EN
+void UART_1_ISR(void)
+{
+    irq_handler(UART_1, UART_1_DEV);
+}
+#endif
+
+static inline void irq_handler(uint8_t uartnum, KINETIS_UART *dev)
+{
+    /*
+    * On Cortex-M0, it happens that S1 is read with LDR
+    * instruction instead of LDRB. This will read the data register
+    * at the same time and arrived byte will be lost. Maybe it's a GCC bug.
+    *
+    * Observed with:
+    * It does not happen with: arm-none-eabi-gcc (4.8.3-9+11)
+    */
+
+    if (dev->S1 & UART_S1_RDRF_MASK) {
+        uint8_t data = dev->D;
+
+        if (config[uartnum].rx_cb != NULL) {
+            config[uartnum].rx_cb(config[uartnum].arg, data);
+        }
+    }
+    else if (dev->S1 & UART_S1_TDRE_MASK) {
+        if (config[uartnum].tx_cb == NULL) {
+            dev->C2 &= ~(UART_C2_TIE_MASK);
+        }
+        else if (config[uartnum].tx_cb(config[uartnum].arg) == 0) {
+            dev->C2 &= ~(UART_C2_TIE_MASK);
+        }
+    }
+
+    if (sched_context_switch_request) {
+        thread_yield();
+    }
+
+    /* clear overrun flag */
+    if (dev->S1 & UART_S1_OR_MASK) {
+#ifndef UART_C4_BRFA
```

Comment:
Move the `#ifndef` outside the `if`
Is `defined(UART_C4_BRFA)` the best way to differentiate between the two varieties of UART modules? 
We should create a macro with a better name to show that this is the intention of the `#ifndef` here and above at `kinetis_set_brfa`